near-sdk/src/store/tree_map/mod.rs

```diff
@@ -101,6 +98,7 @@ where
     K: BorshSerialize,
 {
     root: Option<FreeListIndex>,
+    #[borsh(bound(deserialize = ""))]
     nodes: FreeList<Node<K>>,
```

Comment:
This crate contains a number of unsual types.

E.g. `IndexMap<T>` doesn't technically require any bounds on `T` to perform `BorshSerialize` or `BorshDeserialize`. 
It only requires that there's an implementation of both for [Box<[u8]>](https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/store/index_map.rs#L17), which is provided by `borsh` library. 
Instead, the bounds `T: BorshSerialize`, `T: BorshDeserialize` are baked deep into the type's implementation and basic operations with it, 
[flush, BorshSerialize](https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/store/index_map.rs#L61), [get, BorshDeserialize](https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/store/index_map.rs#L107). 

Due to this pattern being pervasive across the repo, the automatic bound `K: BorshDeserialize` on type parameter in non-skipped field is dropped on commented line with `#[borsh(bound(deserialize = ""))]`, as it wasn't technically required for the trait's implementation in the code previous to this pr. In all similar occasions the bounds were made as close to the version in `master` as possible with these `#[...(bound(...))]` attributes.